src/interpreter/InterpretNot.ts

```diff
@@ -0,0 +1,32 @@
+
+import { Logger } from '../utils';
+import { CommonModel } from '../models/CommonModel';
+import { Schema } from '../models/Schema';
+import { Interpreter, InterpreterOptions } from './Interpreter';
+
+/**
+ * Interpreter function for JSON Schema draft 7 not keyword.
+ *   
+ * @param schema
+ * @param model
+ * @param interpreter
+ */
+export default function interpretNot(schema: Schema, model: CommonModel, interpreter: Interpreter, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions) {
+  if (schema.not === undefined) return;
+  if (typeof schema.not === 'object') {
+    const notSchema = schema.not;
+    const newInterpreterOptions: InterpreterOptions = {
+      ...interpreterOptions, 
+      splitModels: false,
+      allowInheritance: false
+    };
+    const interpretedModels = interpreter.interpret(notSchema, newInterpreterOptions);
+    if (interpretedModels.length > 0) {
+      const interpretedModel = interpretedModels[0];
+      if (interpretedModel.type !== undefined) model.removeType(interpretedModel.type);
+      model.removeEnum(interpretedModel.enum);
```

Comment:
```suggestion
      if (interpretedModel.enum !== undefined) model.removeEnum(interpretedModel.enum);
```

For easier reading the code :) I know that you check statement inside the function, but let's be consistent here (with `interpretedModel.type`) :)

Reply:
True, changed it 🙂

Reply:
Agree 👍